File: 01/caesar.cpp
#include <bits/stdc++.h>

using namespace std;

string encrypt(string word, int key){
    string s = "";
    for(int i=0;i<word.size();i++){
        char c = word[i];
        int j = c - 'a';
        c = (j+key) % 26 + 97;
        s += c;
    }
    return s;
}

string decrypt(string word, int key){
    string s = "";
    for(int i=0;i<word.size();i++){
        char c = word[i];
        int j = c - 'a';
        if(j-key<0){
            c = ((j-key)+26)%26+97;
        } else {
            c = (j-key)%26+97;
        }
        s += c;
    }
    return s;
}

int main()
{
    string word;
    cout<<"Enter the word: ";
    cin>>word;
    int key;
    cout<<"Enter key: ";
    cin>>key;
    string e, d;
    e = encrypt(word, key);
    d = decrypt(e, key);
    cout<<"Original string: "<<word<<endl;
    cout<<"Encrypted string: "<<e<<endl;
    cout<<"Decrypted string: "<<d<<endl;
    return 0;
}

File: 01/Caesar.java
import java.util.Scanner;

public class Caesar {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.print("Enter the word: ");
        String word = in.next();
        System.out.print("Enter key: ");
        int key = in.nextInt();
        String e = encrypt(word, key);
        String d = decrypt(e, key);
        System.out.println("Original string: " + word);
        System.out.println("Encrypted string: " + e);
        System.out.println("Decrypted string: " + d);
    }

    private static String decrypt(String word, int key) {
        StringBuilder s = new StringBuilder();
        for (int i = 0; i < word.length(); i++) {
            char c = word.charAt(i);
            int j = c - 'a';
            if (j - key < 0) {
                c = (char) (((j - key) + 26) % 26 + 97);
            } else {
                c = (char) ((j - key) % 26 + 97);
            }
            s.append(c);
        }
        return s.toString();
    }

    private static String encrypt(String word, int key) {
        StringBuilder s = new StringBuilder();
        for (int i = 0; i < word.length(); i++) {
            char c = word.charAt(i);
            int j = c - 'a';
            c = (char) ((j + key) % 26 + 97);
            s.append(c);
        }
        return s.toString();
    }
}
